lib/core/src/Cardano/Api/Gen.hs

```diff
@@ -708,18 +706,18 @@ genTxOutValue era =
     Left adaOnlyInEra     -> TxOutAdaOnly adaOnlyInEra <$> genLovelace
     Right multiAssetInEra -> TxOutValue multiAssetInEra <$> genValueForTxOut
 
-genTxOut :: CardanoEra era -> Gen (TxOut era)
+genTxOut :: CardanoEra era -> Gen (TxOut ctx era)
 genTxOut era =
   TxOut <$> genAddressInEra era
         <*> genTxOutValue era
         <*> genTxOutDatumHash era
 
-genTxOutDatumHash :: CardanoEra era -> Gen (TxOutDatumHash era)
+genTxOutDatumHash :: CardanoEra era -> Gen (TxOutDatum ctx era)
 genTxOutDatumHash era =
     case scriptDataSupportedInEra era of
-        Nothing -> pure TxOutDatumHashNone
+        Nothing -> pure TxOutDatumNone
         Just supported -> oneof
-            [ pure TxOutDatumHashNone
+            [ pure TxOutDatumNone
             , TxOutDatumHash supported <$> genHashScriptData
```

Comment:
Interesting, it seems there's both `TxOutDatum'` and `TxOutDatumHash` now.

Could we rename `genTxOutDatumHash` to `genTxOutDatum` and also generate `TxOutDatum'`?

Or at the very least add a note about it.

Reply:
I renamed `genTxOutDatumHash` to `genTxOutDatum`, but where is this `TxOutDatum'` ?

Reply:
https://github.com/input-output-hk/cardano-node/blob/4f65fb9a27aa7e3a1873ab4211e412af780a3648/cardano-api/src/Cardano/Api/TxBody.hs#L1136-L1143

I think a note would be good